Intermittent test failer with dstest.SubtestBasicSync

After running the full test overnight, I encounter a failer in the `SubtestBasicSync` of `TestDatastoreSuite`, with the error message `datastore: key not found`.

The fastest way to reproduce this error is replace the line `dstest.SubtestAll(t, replicas[0])` and all checks after it with `dstest.SubtestBasicSync(t, replicas[0])`. Then run `go test -count=10`. The test then fails a few times.

@hsanjuan Thank you for looking into it.

Here is my modifications to TestDatastoreSuite
```go 
func TestDatastoreSuite(t *testing.T) {
	numReplicasOld := numReplicas
	numReplicas = 1
	defer func() {
		numReplicas = numReplicasOld
	}()
	opts := DefaultOptions()
	opts.MaxBatchDeltaSize = 200 * 1024 * 1024 // 200 MB
	replicas, closeReplicas := makeReplicas(t, opts)
	defer closeReplicas()
	//dstest.SubtestAll(t, replicas[0])
	dstest.SubtestBasicSync(t, replicas[0])
	clearDs(t, replicas[0])
	time.Sleep(time.Second)

	for _, r := range replicas {
		q := query.Query{KeysOnly: true}
		results, err := r.Query(q)
		if err != nil {
			t.Fatal(err)
		}
		defer results.Close()
		rest, err := results.Rest()
		if err != nil {
			t.Fatal(err)
		}
		if len(rest) != 0 {
			t.Error("all elements in the suite should be gone")
		}
	}
}

func clearDs(t *testing.T, d ds.Datastore) {
	q, err := d.Query(query.Query{KeysOnly: true})
	if err != nil {
		t.Fatal(err)
	}
	res, err := q.Rest()
	if err != nil {
		t.Fatal(err)
	}
	for _, r := range res {
		if err := d.Delete(ds.RawKey(r.Key)); err != nil {
			t.Fatal(err)
		}
	}
}
```

Here is a few runs of the test
``` bash
xiege@LAPTOP-NPU545OC MINGW64 ~/go/src/github.com/ipfs/go-ds-crdt (master)
$ go test -count=1 -v -run="Suite"
=== RUN   TestDatastoreSuite
--- PASS: TestDatastoreSuite (1.00s)
PASS
ok      github.com/ipfs/go-ds-crdt      1.255s

xiege@LAPTOP-NPU545OC MINGW64 ~/go/src/github.com/ipfs/go-ds-crdt (master)
$ go test -count=1 -v -run="Suite"
=== RUN   TestDatastoreSuite
    TestDatastoreSuite: crdt_test.go:376: datastore: key not found
--- FAIL: TestDatastoreSuite (0.01s)
FAIL
exit status 1
FAIL    github.com/ipfs/go-ds-crdt      0.293s

xiege@LAPTOP-NPU545OC MINGW64 ~/go/src/github.com/ipfs/go-ds-crdt (master)
$ go test -count=1 -v -run="Suite"
=== RUN   TestDatastoreSuite
    TestDatastoreSuite: crdt_test.go:376: datastore: key not found
--- FAIL: TestDatastoreSuite (0.01s)
FAIL
exit status 1
FAIL    github.com/ipfs/go-ds-crdt      0.247s

xiege@LAPTOP-NPU545OC MINGW64 ~/go/src/github.com/ipfs/go-ds-crdt (master)
$ go test -count=1 -v -run="Suite"
=== RUN   TestDatastoreSuite
    TestDatastoreSuite: crdt_test.go:376: datastore: key not found
--- FAIL: TestDatastoreSuite (0.01s)
FAIL
exit status 1
FAIL    github.com/ipfs/go-ds-crdt      0.248s

xiege@LAPTOP-NPU545OC MINGW64 ~/go/src/github.com/ipfs/go-ds-crdt (master)
$ go test -count=1 -v -run="Suite"
=== RUN   TestDatastoreSuite
--- PASS: TestDatastoreSuite (1.00s)
PASS
ok      github.com/ipfs/go-ds-crdt      1.226s
```

--- a/set.go
+++ b/set.go
@@ -88,15 +88,22 @@ func (s *set) Rmv(key string) (*pb.Delta, error) {
 		if r.Error != nil {
 			return delta, err
 		}
-
 		id := strings.TrimPrefix(r.Key, prefix.String())
+		if !ds.RawKey(id).IsTopLevel() {
+			// our prefix matches blocks from other keys i.e. our
+			// prefix is "hello" and we have a different key like
+			// "hello/bye" so we have a block id like
+			// "bye/<block>". If we got the right key, then the id
+			// should be the block id only.
+			continue
+		}
 
-		// check if its already tombed, which case don't add it to the Rmv delta set
+		// check if its already tombed, which case don't add it to the
+		// Rmv delta set.
 		deleted, err := s.inTombsKeyID(key, id)
 		if err != nil {
 			return delta, err
 		}
-
 		if !deleted {
 			delta.Tombstones = append(delta.Tombstones, &pb.Element{
 				Key: key,
@@ -275,7 +282,7 @@ func (s *set) inElemsNotTombstoned(key string) (bool, error) {
 		}
 
 		id := strings.TrimPrefix(r.Key, prefix.String())
-		if !ds.NewKey(id).IsTopLevel() {
+		if !ds.RawKey(id).IsTopLevel() {
 			// our prefix matches blocks from other keys i.e. our
 			// prefix is "hello" and we have a different key like
 			// "hello/bye" so we have a block id like

